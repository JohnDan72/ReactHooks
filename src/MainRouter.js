import React, { useState } from 'react';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import { CustomProvider } from 'rsuite';

// hooks components
import HookApp from "./HookApp";
import CounterApp from './components/01-useState/CounterApp';
import CounterWithCustomHook from './components/01-useState/CounterWithCustomHook';
import SimpleForm from './components/02-useEffect/SimpleForm';
import SimpleFormWithCustomHook from './components/02-useEffect/SimpleFormWithCustomHook';
import MultipleCustomHooks from './components/03-examples/MultipleCustomHooks';
import FocusScreen from './components/04-useRef/FocusScreen';
import RealExampleRef from './components/04-useRef/RealExampleRef';
import Layout from './components/05-layoutEffect/Layout';
import Memorize from './components/06-memo/Memorize';
import MemoHook from './components/06-memo/MemoHook';
import CallbackHook from './components/06-memo/CallbackHook';
import Padre from "./components/07-tareaMemo/Padre";
import TodoApp from './components/08-useReducer/TodoApp';

// notfound
import NotFound from './components/NotFound';

// context components
import ContextApp from './components/09-useContext/ContextApp';
import HomeScreen from './components/09-useContext/HomeScreen';
import AboutScreen from './components/09-useContext/AboutScreen';
import LoginScreen from './components/09-useContext/LoginScreen';
import { UserContext } from './components/09-useContext/UserContext';


// data context
// import data from './userData.json';


const MainRouter = () => {
    const [user, setUser] = useState({})
    const [theme, setTheme] = useState('dark');

    return (
        <UserContext.Provider value={{
            user, setUser
        }}>
            <CustomProvider theme={theme}>
                <Router>
                    <Routes>
                        <Route path="/hooks" element={<HookApp setTheme={setTheme} />} >
                            <Route path="" element={<h2>Home</h2>} />
                            <Route path="counter" element={<CounterApp />} />
                            <Route path="counterHook" element={<CounterWithCustomHook />} />
                            <Route path="useEffect" element={<SimpleForm />} />
                            <Route path="useEffectHook" element={<SimpleFormWithCustomHook />} />
                            <Route path="useFetch" element={<MultipleCustomHooks />} />
                            <Route path="useRef" element={<FocusScreen />} />
                            <Route path="useRefReal" element={<RealExampleRef />} />
                            <Route path="useLayoutEffect" element={<Layout />} />
                            <Route path="memorize" element={<Memorize />} />
                            <Route path="memoHook" element={<MemoHook />} />
                            <Route path="useCallback" element={<CallbackHook />} />
                            <Route path="tareaMemo" element={<Padre />} />
                            <Route path="useReducer" element={<TodoApp />} />
                        </Route>
                        <Route path="/context" element={<ContextApp setTheme={setTheme} />}>
                            <Route path="" element={<HomeScreen />} />
                            <Route path="about" element={<AboutScreen />} />
                            <Route path="login" element={<LoginScreen />} />
                        </Route>
                        <Route path="*" element={<NotFound />} />
                    </Routes>
                </Router>
            </CustomProvider>
        </UserContext.Provider>
    );
}

export default MainRouter;