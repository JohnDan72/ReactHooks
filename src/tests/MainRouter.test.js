import { mount, shallow } from "enzyme"
import MainRouter from "../MainRouter";

describe('Pruebas Main Router', () => {
    
    const wrapper = shallow(<MainRouter />);

    test('debe renderizar el component', () => {
        
        expect(wrapper).toMatchSnapshot();
    })
    
})
