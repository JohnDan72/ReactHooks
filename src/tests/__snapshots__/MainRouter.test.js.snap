// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas Main Router debe renderizar el component 1`] = `
<ContextProvider
  value={
    Object {
      "setUser": [Function],
      "user": Object {},
    }
  }
>
  <CustomProvider
    theme="dark"
  >
    <BrowserRouter>
      <Routes>
        <Route
          element={
            <HookApp
              setTheme={[Function]}
            />
          }
          path="/hooks"
        >
          <Route
            element={
              <h2>
                Home
              </h2>
            }
            path=""
          />
          <Route
            element={<CounterApp />}
            path="counter"
          />
          <Route
            element={<CounterWithCustomHook />}
            path="counterHook"
          />
          <Route
            element={<SimpleForm />}
            path="useEffect"
          />
          <Route
            element={<SimpleFormWithCustomHook />}
            path="useEffectHook"
          />
          <Route
            element={<MultipleCustomHooks />}
            path="useFetch"
          />
          <Route
            element={<FocusScreen />}
            path="useRef"
          />
          <Route
            element={<RealExampleRef />}
            path="useRefReal"
          />
          <Route
            element={<Layout />}
            path="useLayoutEffect"
          />
          <Route
            element={<Memorize />}
            path="memorize"
          />
          <Route
            element={<MemoHook />}
            path="memoHook"
          />
          <Route
            element={<CallbackHook />}
            path="useCallback"
          />
          <Route
            element={<Padre />}
            path="tareaMemo"
          />
          <Route
            element={<TodoApp />}
            path="useReducer"
          />
        </Route>
        <Route
          element={
            <ContextApp
              setTheme={[Function]}
            />
          }
          path="/context"
        >
          <Route
            element={<HomeScreen />}
            path=""
          />
          <Route
            element={<AboutScreen />}
            path="about"
          />
          <Route
            element={<LoginScreen />}
            path="login"
          />
        </Route>
        <Route
          element={<NotFound />}
          path="*"
        />
      </Routes>
    </BrowserRouter>
  </CustomProvider>
</ContextProvider>
`;
